_books/ion-1-1/src/macros/system_macros.md

```diff
@@ -223,35 +249,47 @@ Then:
 (:foo_struct c 3) => { foo_a: 1, foo_b: 2, foo_c: 3 }
 ```
 
+For normative examples, see [`make_struct`](https://github.com/amazon-ion/ion-tests/tree/main/conformance/system_macros/make_struct.ion) in the Ion conformance test suite.
+
 #### `make_decimal`
 
 ```ion
-(macro make_decimal (flex_int::coefficient flex_int::exponent) /* Not representable in TDL */)
+(macro make_decimal (coefficient exponent) /* Not representable in TDL */)
 ```
 
 This is no more compact than the regular binary encoding for decimals.
 However, it can be used in conjunction with other macros, for example, to represent fixed-point numbers.
 
+Both `coefficient` and `exponent` must be (or evaluate to) a single integer value. 
+
 ```ion
 (macro usd (cents) (.annotate USD (.make_decimal cents -2))
 
 (:usd 199) =>  USD::1.99
 ```
 
+> [!NOTE]
+> It is not possible to use `make_decimal` to construct any negative zero value because Ion integers do not have signed zero.
+
+For normative examples, see [`make_decimal`](https://github.com/amazon-ion/ion-tests/tree/main/conformance/system_macros/make_decimal.ion) in the Ion conformance test suite.
+
 #### `make_timestamp`
 
 ```ion
-(macro make_timestamp (uint16::year
-                       uint8::month?
-                       uint8::day?
-                       uint8::hour?
-                       uint8::minute?
-                       /*decimal*/ second?
-                       int16::offset_minutes?) /* Not representable in TDL */)
+(macro make_timestamp (year month? day? hour? minute? second? offset_minutes?) /* Not representable in TDL */)
 ```
 Produces a non-null, unannotated timestamp at various levels of precision.
 When `offset` is absent, the result has unknown local offset; offset `0` denotes UTC.
+
 The arguments to this macro may not be any null value.
+The evaluated argument for the `second` parameter, if present, must be a decimal or integer value that is greater than
+or equal to zero and less than 60. The evaluated arguments for all other parameters, if present, must be integer values.
+
+The `make_timestamp` macro has additional rules (that cannot be expressed in the macro signature) because it must
+construct a valid Ion timestamp value.
+The `offset_minutes` and `hour` parameters may only be present if `minute` is present. Aside from `offset_minutes`, if 
+any evaluated argument is present, the evaluated arguments for all parameters to the left must also be present.
+The precision of the constructed timestamp is determined by which parameters have non-empty arguments.
```

Comment:
Why specify the range constraints for `second` but not for `day`, `month`, or `hour`?  
Secondly, could one not default unspecified parameters to `0`? Or should that be left up to the user to define such macros if they are useful?

Reply:
> Why specify the range constraints for second but not for day, month, or hour?

I'll update the description.

> Secondly, could one not default unspecified parameters to 0?

Well, that already happens (somewhat) because of timestamp semantics. E.g. these macro invocations all represent the same point in time, but with different precisions.
```
(:make_timestamp 2024 12) 
(:make_timestamp 2024 12 1) 
(:make_timestamp 2024 12 1 0 0) 
(:make_timestamp 2024 12 1 0 0 0) 
(:make_timestamp 2024 12 1 0 0 0.000) 
```